profiling/src/profile/internal/labels/label_set.rs

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::super::*;
+
+/// A canonical representation for sets of labels.
+/// You should only use the impl functions to modify this.
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct LabelSet {
+    sorted_labels: Box<[LabelId]>,
+}
+
+impl LabelSet {
+    pub fn iter(&self) -> core::slice::Iter<'_, LabelId> {
+        self.sorted_labels.iter()
+    }
+
+    pub fn new(mut v: Vec<LabelId>) -> Self {
+        v.sort_unstable();
+        let sorted_labels = v.into_boxed_slice();
+        Self { sorted_labels }
+    }
```

Comment:
I wonder how often `v` is the correct size already. It _may_ be prudent to take a `Box<[LabelId]>` in the constructor instead, so the caller is more aware of making a correctly sized slice in the first place, instead of having excess capacity which will then trigger re-allocation.

Reply:
However, if we move to using an arena, it will may cause a re-allocation anyway. So not high priority.

Reply:
I'll leave that for a later PR